src/lazy/components/client.tsx

```diff
@@ -38,21 +38,20 @@ export function createComponentClient<C extends ComponentType<any>>({
 
     useMemo(() => {
       if (isOwnPhase)
-        deferred
-          .start()
-          .then(() => setFallback(null))
```

Comment:
Removed `setFallback(null)` as it is now handled by the Suspense rendered fallback itself: when that gets unmounted it nullifies it automatically, so we can properly handle promises thrown immediately after the component has been loaded, still maintaining SSR